AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/RemoteSyncEngine+AuthModeStrategyDelegate.swift

```diff
@@ -33,19 +33,6 @@ extension RemoteSyncEngine: AuthModeStrategyDelegate {
             return false
         }
 
-        // Note: blocking is not recommended
-        let group = DispatchGroup()
```

Comment:
Could this have been a DispatchSemaphore instead of a DispatchGroup? It looks like we are only calling group.enter() once

Reply:
Yes, it could have been.  With that said, I actually think a `DispatchGroup` was better here (and would have been better most places a semaphore was used in our codebase).  The `DispatchGroup` is basically for waiting for one or more threads to be done.  The `DispatchSemaphore` can be used for that as well (and frequently was within the Amplify codebase), but is more general and can also be used to allow a limited number of threads to continue to execute simultaneously.  Given that the intended behavior was the former, I would have used a `DispatchGroup`, but again, either would work.